evap/results/exporters.py

```diff
@@ -122,46 +134,49 @@ def filter_evaluations(semesters, evaluation_states, degrees, course_types, cont
                 continue
             if not evaluation.can_publish_rating_results and not include_not_enough_voters:
                 continue
-            results = OrderedDict()
+            results: OrderedDict[EvaluationResult, list[RatingResult]] = OrderedDict()
             for contribution_result in get_results(evaluation).contribution_results:
                 for questionnaire_result in contribution_result.questionnaire_results:
                     # RatingQuestion.counts is a tuple of integers or None, if this tuple is all zero, we want to exclude it
+                    question_results: list[RatingResult] = questionnaire_result.question_results  # type: ignore[assignment]
                     if all(
                         not question_result.question.is_rating_question
                         or question_result.counts is None
                         or sum(question_result.counts) == 0
-                        for question_result in questionnaire_result.question_results
+                        for question_result in question_results
                     ):
                         continue
                     if (
                         not contributor
                         or contribution_result.contributor is None
                         or contribution_result.contributor == contributor
                     ):
-                        results.setdefault(questionnaire_result.questionnaire.id, []).extend(
-                            questionnaire_result.question_results
-                        )
+                        results.setdefault(questionnaire_result.questionnaire.id, []).extend(question_results)
                         used_questionnaires.add(questionnaire_result.questionnaire)
-            evaluation.course_evaluations_count = evaluation.course.evaluations.count()
-            if evaluation.course_evaluations_count > 1:
+            evaluation.course_evaluations_count = evaluation.course.evaluations.count()  # type: ignore[attr-defined]
```

Comment:
Can we declare `evaluation` to be `Any` once inside this function that does the monkey patching, instead of adding multiple `type ignore` comments?

Reply:
Yes we could do that. But I think some type ignores are better than `Any` to cover a little bit and think about what and why is monkeypatched.

Reply:
To me, `Any` says that this variable currently isn't typechecked, so it tells you that you can not rely on typechecking. Annotating as `Evaluation`, on the other hand, would imply that it is treated according to static typing rules and typechecked, which is a lie here.

To make this code typecheck in the future
* with `Any`, we'd have to change one place (replace `Any` with a type that also allows the monkeypatched attributes)
* with lots of `type: ignore`, we have to go through lots of code, and find out for each annotation again whether it was due to a bug/failure in mypy, problem with the stubs, or because we lied to the typechecker.

I have a strong opinion here. The code doesn't treat `evaluation` according to static type checking rules for `Evaluation`, so we shouldn't make the type checker assume that it does and then go on and ignore all violations.

@niklasmohrin opinions?

Reply:
Yeah, let's use `Any`